content/en/opentelemetry/agent/install_agent_with_collector.md

```diff
@@ -0,0 +1,573 @@
+   {{< /code-block >}}
+   <div class="alert alert-info"><p>It is required to set the <code>hostPort</code> in order for the container port to be exposed to the external network. This enables configuring the OTLP exporter to point to the IP address of the node to which the <code>otel-agent</code> is assigned.</p><p>Alternatively, if you doesn't want to expose the port, you can use the agent service instead via <code>&lt;SERVICE_NAME&gt;.&lt;SERVICE_NAMESPACE&gt;.svc.cluster.local</code>.</p></div>
+1. (Optional) Enable additional Datadog features.
+   <div class="alert alert-danger">Enabling these features may incur additional charges. Review the <a href="https://www.datadoghq.com/pricing/"> pricing page</a> and talk to your CSM before proceeding.</div>
+   {{< code-block lang="yaml" filename="datadog-values.yaml" collapsible="true" >}}
+datadog:
+  ...
+  apm:
+    portEnabled: true
+    peer_tags_aggregation: true
+    compute_stats_by_span_kind: true
+    peer_service_aggregation: true
+  orchestratorExplorer:
+    enabled: true
+  processAgent:
+    enabled: true
+    processCollection: true
+   {{< /code-block >}}
+
+{{% collapse-content title="Completed datadog-values.yaml file" level="p" %}}
+Your `datadog-values.yaml` file should look something like this:
+{{< code-block lang="yaml" filename="datadog-values.yaml" collapsible="false" >}}
+agents:
+  image:
+    repository: datadog/agent-dev
+    tag: nightly-ot-beta-main-jmx
+    doNotCheckTag: true
+
+datadog:
+  apiKeyExistingSecret: datadog-secret
+  appKeyExistingSecret: datadog-secret
+
+  otelCollector:
+    enabled: true
+    ports:
+      - containerPort: "4317"
+        hostPort: "4317"
+        name: otel-grpc
+      - containerPort: "4318"
+        hostPort: "4318"
+        name: otel-http
+  apm:
+    portEnabled: true
+    peer_tags_aggregation: true
+    compute_stats_by_span_kind: true
+    peer_service_aggregation: true
+  orchestratorExplorer:
+    enabled: true
+  processAgent:
+    enabled: true
+    processCollection: true
+
+  podLabelsAsTags:
+    app: kube_app
+    release: helm_release
+   {{< /code-block >}}
+
+{{% /collapse-content %}}
+
+### Configure the OpenTelemetry Collector
+
+The DataDog Helm chart provides a sample OpenTelemetry Collector configuration that's a great starting point for most projects. This section walks you through the predefined pipelines and included OpenTelemetry components.
+
+This is the full OpenTelemetry Collector configuration in `otel-config.yaml`:
+
+{{< code-block lang="yaml" filename="otel-config.yaml" disable_copy="false" collapsible="true" >}}
+receivers:
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: "otelcol"
+          scrape_interval: 10s
+          static_configs:
+            - targets: ["0.0.0.0:8888"]
+  otlp:
+    protocols:
+      grpc:
+         endpoint: 0.0.0.0:{{ include "get-port-number-from-name" (dict "ports" .Values.datadog.otelCollector.ports "portName" "otel-grpc") }}
```

Comment:
We shouldn't have helm template funcs outside of Helm. We should hard code the port

Reply:
Updated to hard code the values:

```
  otlp:
    protocols:
      grpc:
         endpoint: 0.0.0.0:4317
      http:
         endpoint: 0.0.0.0:4318
```